### ISSUE TITLE
Microphone button not stopping

### ISSUE THREAD

**Issue by Lucas-almma** [microsoft/azurechat#383]
Hello, I'm running into a problem. When I activate the microphone to dictate to Azure, dictation works just fine.

But when I click the microphone again to stop it, it doesn't stop. Has anyone experienced this bug?

I tried using Chrome, FF, and Edge with the same results.

Any guidance or help will be appreciated.

**Comment by aliADSP** [comment]
@Lucas-almma Yes I think the mic is broken. They need to use a library to manage the mic state across browsers instead of a home cooked version they have

**Comment by Lucas-almma** [comment]
Thank you for your reply. I have since moved away from msft/azurechat to my
own multi-LLM platform.

ᐧ

**Comment by aliADSP** [comment]
Makes sense. This is still needs a lot of work. The backend needs to be separated out from the API layer. There needs to be agonistic model handling, monitoring and load balancing, the chat titles need to be auto-renamed. add rag index must be easier than what it is. etc. Also they need to move away from doc intelligence. Multimodal models can respond much better with original document passed in.
